src/core/DataSourceBase.ts

```diff
@@ -72,8 +72,8 @@ export abstract class DataSourceBase<TQuery extends DataQuery, TOptions extends
     return this.fetch<T>({ method: 'GET', url, params });
```

Comment:
I don't see tests added to the data sources that cover the desired behavior or documentation describing the pattern.

My concern is if additional data sources are added:

- how are authors expected to know that some calls should hide show error and some should not?
- is there documentation describing how it should be validated? Either with tests or manual validation?
- how do I make sure I don't suppress errors unexpectedly?